### PR TITLE
Docs for cached array and text blocks.

### PR DESCRIPTION
**Pull request by ganicke**
- [x] Adjust help paths for cached blocks in _blocks.ts_.
- [x] Add or expand remaining **Text** and **Arrays** topics.

Fixes https://github.com/Microsoft/pxt-adafruit/issues/289.

### CODE REVIEW

File: common-docs/reference/arrays/get.md [hunk 1]
```diff
@@ -1,7 +1,23 @@
...
+Get a value from an array at a particular index.
 
 ```block
-["hello"][0]
-```
\ No newline at end of file
+let item = [""][0]
+```
+
+## Returns
```
Comment (pelikhan) [inline common-docs/reference/arrays/get.md]: Add section on "out of bounds" issues

**Comment by ganicke** [comment]
Not sure how/if we should discuss negative indexes. Code like this works but array length only counts the items with a positive index:

```typescript
let list: string[] = []
list = ["fafasdf", "ertertwe", "dfgsg"]
list[-8] = "afsdf"
basic.showString(list[-1])
basic.showNumber(list.length)
basic.showString(list[-8])
basic.showNumber(list.indexOf("afsdf"))
```

**Comment by pelikhan** [comment]
@ganicke do you need to bring some of those doc updates to master?

**Comment by ganicke** [comment]
Indeed. Did it here first to address a specific issue filed but I'll do a PR for ``master`` also.

**Pull request merged**